src/components/common/icons/KeyholeIcon/index.tsx

```diff
@@ -1,9 +1,32 @@
-import Keyhole from '@/components/common/icons/KeyholeIcon/keyhole.svg'
+import css from '@/components/common/icons/CircularIcon/styles.module.css'
+import LockIcon from '@/public/images/common/lock.svg'
+import { Badge, SvgIcon } from '@mui/material'
 
-import CircularIcon from '../CircularIcon'
-
-const KeyholeIcon = ({ size = 40 }: { size?: number }) => {
-  return <CircularIcon icon={Keyhole} badgeColor="error" size={size} />
+const KeyholeIcon = ({ size = 28 }: { size?: number }) => {
+  return (
+    <Badge
+      color="error"
+      overlap="circular"
+      variant="dot"
+      anchorOrigin={{
+        vertical: 'bottom',
+        horizontal: 'right',
+      }}
+      className={css.badge}
+    >
+      <SvgIcon
+        component={LockIcon}
+        inheritViewBox
+        sx={{
+          height: size,
+          width: size,
+          '& path': {
+            fill: ({ palette }) => palette.border.main,
```

Comment:
We could also set fill=currentColor in the SVG and use the `color` CSS property.